src/main/java/mat/dao/clause/impl/MeasureDAOImpl.java

```diff
@@ -854,6 +874,29 @@ public ShareLevel findShareLevelForUser(String measureId, String userID, String
         return shareLevel;
     }
 
+    @Override
+    public Map<String, String> findShareLevelsForUser(String userID, Set<String> measureSetIds) {
+        logger.debug("MeasureDAOImpl::findShareLevelsForUser - enter");
+        final StopWatch stopWatch = new StopWatch();
+        stopWatch.start();
+
+        final Session session = getSessionFactory().getCurrentSession();
+        NativeQuery query = session.createNativeQuery(SELECT_SHARE_LEVELS);
+        query.setParameter("userId", userID);
+        query.setParameter("measureSetIds", measureSetIds);
+        List<Object[]> list = query.getResultList();
+
+        Map<String, String> result = new HashMap<>();
+
+        list.stream().forEach(row -> {
+            result.put(String.valueOf(row[0]), String.valueOf(row[1]));
+        });
```

Comment:
This could be done entirely in streams, though I don't know if it would be any more performant.
```
 Map<String, String> result = list.stream().collect(toMap(row -> String.valueOf(row[0]), row -> String.valueOf(row[1])));
```

Reply:
right.. just a code preference